{% extends "base.html" %}
{% block page_content %}
  <div id="outer-content">
    <div id="content">
      <div id="main">
	<div class="header">
	<h1>About us</h1>
	</div>
<p>This resource was built and is maintained by the Kroncke lab at the Vanderbilt University Medical Center
    <a href="https://kroncke-lab.github.io/index.html">(Kroncke lab website)</a>. Please send any questions or comments to
    brett.m.kroncke.1@vumc.org.</p>

<div class="header">
	<h1>Background</h1>
	</div>
<p> 
Sequencing an individual’s genome now costs less than many routine medical procedures. A resulting
          vision is that everyone will have their genome sequenced early in life to enable individualized medical advice
          about disease prevention and drug selection. A major concern with this vision, however, is proper
          interpretation of the overwhelming volume of discovered novel and rare variants. In other contexts, a new
          diagnostic test can be benchmarked and validated in studies that compare large populations with and without a
          disorder to determine the predictive value of a positive result. In the genetic sequencing context, however,
          a positive test for most variants cannot be applied to enough heterozygous individuals for a definitive
          association with disease. A major resulting challenge facing contemporary genomic medicine is the clinical community’s
          desire for yes/no
          answers to the nuanced issue of whether a specific genetic variant will produce a meaningful phenotype.
          The current framework used to assess the significance of these variants classifies them from likely pathogenic/
          pathogenic to likely benign/benign, with most stuck as variant of uncertain significance. Here we present a
          data-driven estimate of disease penetrance along side raw data for interpreting variants in KCNQ1, KCNH2, and SCN5A.
          To estimate penetrance of disease, heuristically, we found that the innate diagnostic information one learns
          about a variant from three-dimensional variant location, <i>in vitro</i> functional data, and <i>in silico</i>
          predictors is equivalent to the diagnostic information one learns about that same variant by clinically
          phenotyping around 10-20 heterozygotes. These results are published in 2020 in
    <a href="https://pubmed.ncbi.nlm.nih.gov/32569262/">PLOS Genetics</a> with an additional manuscript in review
    (<a href="https://github.com/kroncke-lab/Bayes_KCNH2_LQT2_Penetrance">github site</a>).
    We present the results from these analyses in the form of searchable
          tables.


    </div>
  </div>
</div>
{% endblock %}

