<!--
TODO: 2) Remove duplicate variants displayed in distances table
-->
{% extends "base.html" %}
{% load static %}
{% block page_content %}

<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.1.36/vfs_fonts.js"></script>

<script type="text/javascript" src="https://code.jquery.com/jquery-3.5.1.js"></script>
<script type="text/javascript" src="https://cdn.datatables.net/v/ju/dt-1.10.24/b-1.7.0/b-colvis-1.7.0/b-html5-1.7.0/date-1.0.3/r-2.2.7/sc-2.0.3/sb-1.0.1/sl-1.3.3/datatables.min.js"></script>


<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.10.24/css/jquery.dataTables.min.css"/>


<div align="center" class="jumbotron text-center">
    <h1>
        <p> <i>SCN5A</i> Variant <b>{{ variant.var }}</b> Detail </p>
    </h1>
</div>

<i>SCN5A</i> {{ variant.var }} was found in a total of {{variant.total_carriers}} carriers in {{ recs_clin| length }} papers:
{{variant.brs1}} had Brugada syndrome,{{variant.lqt3}} had LQTS .

{% if variant.gnomad > 0 %}
    {{variant.var}} is present in {{variant.gnomad}} alleles in gnomAD.
{% else %}
    {{variant.var}} is not present in gnomAD.
{% endif %}
{{variant.var}} has been functionally characterized in {{ recs_funcs| length }} papers.
This residue is located in a {{variant.structure}} region for LQT2.

{% if alpha_brs1 != "NA" %}
<i>In silico</i> predictions, functional data (if available), and location in structure are equivalent to observing
{{alpha_brs1}} individuals with Brugada syndrome, {{ alpha_lqt3 }} with LQTS, and {{beta}} unaffected individuals.
{% endif %}
Missense variants are combined across degenerate codon substitutions since these data were not consistently available.

<body>

<div align="center" class="content">
    <h2>
        <p> <i>In Silico</i> Data </p>
    </h2>
</div>
<table class="table table-striped table-bordered">
    <thead>
    <tr>
        <th>PROVEAN</th>
        <th>PolyPhen-2</th>
        <th>BLAST-PSSM</th>
        <th>REVEL</th>
        <th>Penetrance Density BrS (%)</th>
        <th>Penetrance Density LQT3 (%)</th>
    </tr>
    </thead>
    <tbody>
    <tr>
        <td>{{variant.provean_score}}</td>
        <td>{{variant.pph2_prob}}</td>
        <td>{{variant.blast_pssm}}</td>
        <td>{{variant.revel_score}}</td>
        <td>{% widthratio variant.brs1_dist 1 100 %}</td>
        <td>{% widthratio variant.lqt3_dist 1 100 %}</td>
    </tr>
   </tbody>
</table>
PROVEAN scores less than -2 are considered deleterious. REVEL scores higher than 0.5 or 0.75 is considered
likely pathogenic (higher sensitivity with the former cutoff, higher specificity with the latter cutoff).
A PolyPhen-2 score of 0.85 or greater is considered likely pathogenic.
BLAST-PSSM reflects the evolutionary conservation of residue substitutions, more negative numbers indicate
fewer observations of the specific substitution than is expected. Penetrance Density is our previously published method
to calculate the average BrS/LQTS probability density in a shell of residues surrounding a residue of interest
(<a href="https://pubmed.ncbi.nlm.nih.gov/30828412/">ref</a>).
<p></p>

<div align="center" class="content">
    <h2>
        <p> Reported Carrier Data </p>
    </h2>
</div>
<table id="clinrecs" class="table table-striped table-bordered">
    <thead>
    <tr>
        <th>PubMed ID</th>
        <th>Year</th>
        <th>Carriers</th>
        <th>Unaffected</th>
        <th>LQT3</th>
        <th>BrS1</th>
        <th>Other</th>
        <th>Other Disease</th>
    </tr>
    </thead>
    <tbody>
    {% if clin_papers > 0 %}
        {% for p in recs_clin %}
    <tr>
        {% if p.pmid != "Italy Cohort" and p.pmid != "Japan Cohort" and p.pmid != "Paris Cohort" %}
        <td><a href="https://pubmed.ncbi.nlm.nih.gov/{{ p.pmid }}">{{ p.pmid }}</a></td>
        {% else %}
        <td>{{ p.pmid }}</td>
        {% endif %}
        <td>{{p.year}}</td>
        <td>{{p.total_carriers}}</td>
        <td>{{p.unaff_gnomad}}</td>
        <td>{{p.lqt3}}</td>
        <td>{{p.brs1}}</td>
        <td>{{p.other}}</td>
        <td>{{p.other_diag}}</td>
    </tr>
        {% endfor %}
    {% endif %}
    <tr>
        <td>LITERATURE, COHORT, AND GNOMAD:</td>
        <td>-</td>
        <td>{{variant.total_carriers}}</td>
        <td>{{variant.unaff_gnomad}}</td>
        <td>{{variant.lqt3}}</td>
        <td>{{variant.brs1}}</td>
        <td>{{variant.other}}</td>
        <td>-</td>
    </tr>
    <tr>
        <td>VARIANT FEATURES ALONE:</td>
        <td>-</td>
        <td>15</td>
        <td>{{beta}}</td>
        <td>{{alpha_lqt3}}</td>
        <td>{{alpha_brs1}}</td>
        <td>-</td>
        <td>-</td>
    </tr>
   </tbody>
</table>
Summary totals might not agree with the literature table because of duplicate patients, which were excluded from the
total counts.

<p></p>
{% if func_papers > 0 %}
<div align="center" class="content">
    <h2>
        <p>Functional Data</p>
    </h2>
</div>
Peak and late/persistent current are relative to wildtype (100% being no different from wildtype).
V<sub>0.5</sub> act/inact are the voltages at which half of the maximal current is reached during an activation and
inactivation protocol, each is in units of mV and relative to wildtype.

<table id="funcrecs" class="table table-striped table-bordered">
    <thead>
    <tr>
        <th>PubMed ID</th>
        <th>Year</th>
        <th>Cell Type</th>
        <th>Peak Current (%WT)</th>
        <th>V<sub>1/2</sub> Act. (mV)</th>
        <th>V<sub>1/2</sub> Inact. (mV)</th>
        <th>Late/Persistent Current (%WT)</th>
    </tr>
    </thead>
    <tbody>
    {% for p in recs_funcs %}
    <tr>
        <td><a href="https://pubmed.ncbi.nlm.nih.gov/{{ p.pmid }}">{{ p.pmid }}</a></td>
        <td>{{ p.year }}</td>
        <td>{{p.cell_type}}</td>
        <td>{% widthratio p.peak 1 100 %}</td>
        <td>{{p.vhalf_act}}</td>
        <td>{{p.vhalf_inact}}</td>
        <td>{% widthratio p.late 1 100 %}</td>
    </tr>
    {% endfor %}
   </tbody>
</table>
{% endif %}
<p></p>

{% if var_type %}
<div align="center" class="content">
    <h2>
        <p> <b>{{ variant.var }}</b> has {{ recs_dists| length }} previously observed neighbors within 15 angstroms </p>
    </h2>
</div>
A residue within a folded protein on average has nearest neighbors that fall roughly into two shells: a "nearest"
neighbor around 5-6 angstroms and a second shell around 11 angstroms. NOTE: some residues appear multiple times at
different distances. This results from the fact that the functional Kv11.1 channel is a homotetramer and occasionally
the same residue from multiple subunits is present within the 15A window. All variants shown in the rightmost
column have been observed in at least one individual in the literature or gnomAD.
<p></p>
<table id="distrecs" class="table table-striped table-bordered">
    <thead>
    <tr>
        <th>Neighbor</th>
        <th>Distance (Angstroms)</th>
        <th>Variants Observed in Individuals</th>
    </tr>
    </thead>
    <tbody>
    {% for p in recs_dists %}
        <tr>
            <td>{{p.neighbor}}</td>
            <td>{% widthratio p.distance 1 1 %}</td>
            <td>
            {% for r in recs_vars %}
                {% if r.resnum == p.neighbor %}
                         <a href="{% url 'scn5a:detail' r.hgvsc %}">{{r.var}}, </a>
                {% endif %}
            {% endfor %}
            </td>
        </tr>
    {% endfor %}

   </tbody>
</table>
{% endif %}

</body>
<p></p>
<p></p>
<p></p>
<script>
// Variant Detail Tables:
$(document).ready(function() {
    var table = $('#clinrecs').DataTable( {
        scrollY: 300,
        scrollCollapse: true,
        paging: false,
        searching: false, info: false
        });
        $('#funcrecs').DataTable( {
        scrollY: 300,
        scrollCollapse: true,
        paging: false,
        searching: false, info: false
        });
        $('#distrecs').DataTable( {
        scrollY: 300,
        scrollCollapse: true,
        paging: false,
        searching: false, info: false
        });
});
</script>
{% endblock %}