{% extends "base.html" %}
{% load static %}

{% block extra_css %}
    {{ block.super }}
    <style>
    .lds-dual-ring {
      display: flex;
        justify-content: center;
        align-items: center;
    }
    .lds-dual-ring:after {
      content: " ";
      display: block;
      width: 64px;
      height: 64px;
      margin: 28px;
      border-radius: 50%;
      border: 6px solid #78cbf1;
      border-color: #646464 transparent #646464 transparent;
      animation: lds-dual-ring 1.2s linear infinite;
    }
    @keyframes lds-dual-ring {
      0% {
        transform: rotate(0deg);
      }
      100% {
        transform: rotate(360deg);
      }
    }
    div.fa-beer {
        margin-top: 70px;
    }
    </style>
    <!-- DataTables 1.x Core CSS -->
    <link rel="stylesheet" href="https://cdn.datatables.net/1.13.8/css/jquery.dataTables.min.css">
    <!-- Buttons Extension CSS (v2.x for DT1) -->
    <link rel="stylesheet" href="https://cdn.datatables.net/buttons/2.4.2/css/buttons.dataTables.min.css">
    <!-- Select Extension CSS (v1.x for DT1) -->
    <link rel="stylesheet" href="https://cdn.datatables.net/select/1.7.0/css/select.dataTables.min.css">
    <!-- SearchBuilder Extension CSS (v1.x for DT1) -->
    <link rel="stylesheet" href="https://cdn.datatables.net/searchbuilder/1.6.0/css/searchBuilder.dataTables.min.css">
    <!-- SearchPanes Extension CSS (v2.x for DT1) -->
    <link rel="stylesheet" href="https://cdn.datatables.net/searchpanes/2.2.0/css/searchPanes.dataTables.min.css">
    <!-- Scroller Extension CSS (virtual scrolling) -->
    <link rel="stylesheet" href="https://cdn.datatables.net/scroller/2.3.0/css/scroller.dataTables.min.css">
{% endblock %}

{% block extra_js %}
    {{ block.super }}
    <!-- DataTables 1.x Core JS -->
    <script src="https://cdn.datatables.net/1.13.8/js/jquery.dataTables.min.js"></script>
    <!-- JSZip (required for Excel export) -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jszip/3.10.1/jszip.min.js"></script>
    <!-- pdfmake (required for PDF export) -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.2.7/pdfmake.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.2.7/vfs_fonts.js"></script>
    <!-- Buttons Extension JS (v2.x for DT1) -->
    <script src="https://cdn.datatables.net/buttons/2.4.2/js/dataTables.buttons.min.js"></script>
    <script src="https://cdn.datatables.net/buttons/2.4.2/js/buttons.html5.min.js"></script>
    <script src="https://cdn.datatables.net/buttons/2.4.2/js/buttons.colVis.min.js"></script>
    <!-- Select Extension JS (v1.x for DT1) -->
    <script src="https://cdn.datatables.net/select/1.7.0/js/dataTables.select.min.js"></script>
    <!-- SearchBuilder Extension JS (v1.x for DT1) -->
    <script src="https://cdn.datatables.net/searchbuilder/1.6.0/js/dataTables.searchBuilder.min.js"></script>
    <!-- SearchPanes Extension JS (v2.x for DT1) -->
    <script src="https://cdn.datatables.net/searchpanes/2.2.0/js/dataTables.searchPanes.min.js"></script>
    <!-- Responsive Extension JS -->
    <script src="https://cdn.datatables.net/responsive/2.5.0/js/dataTables.responsive.min.js"></script>
    <!-- Scroller Extension JS (virtual scrolling for large datasets) -->
    <script src="https://cdn.datatables.net/scroller/2.3.0/js/dataTables.scroller.min.js"></script>
    <!-- Plotly -->
    <script src='https://cdn.plot.ly/plotly-2.27.1.min.js'></script>
    <!-- Shared Chart Theme -->
    <script type="text/javascript" src="{% static 'js/chart_theme.js' %}"></script>
    <!-- Custom SCN5A JS -->
    <script type="text/javascript" src="{% static 'js/var_table_scn5a.js' %}"></script>
    <!-- Initialize Bootstrap tooltips -->
    <script>
        document.addEventListener('DOMContentLoaded', function() {
            var tooltipTriggerList = [].slice.call(document.querySelectorAll('[data-bs-toggle="tooltip"]'));
            tooltipTriggerList.map(function(el) {
                return new bootstrap.Tooltip(el);
            });
        });
    </script>
{% endblock %}

{% block page_content %}

<div align="center" class="text-center" style="margin-top: 30px">
    <h1>
        <p><i>SCN5A</i> Variant Browser </p>
    </h1>
</div>
<div>
    <p>
        Here we have a comprehensive list of variants in <i>SCN5A</i> found in individuals affected and unaffected with the
        two diseases most commonly associated with variants in SCN5A: Brugada Syndrome (BrS1; OMIM: 601144) and
        long QT type 3 (LQT3; OMIM: 603830). Coding variants are referenced to isoform Q14524-1 (ENST00000333535.9) to
        match convention. 'Total Carriers' is the number of heterozygotes diagnosed with LQT3, BrS1, Unaffected with
        disease, and found in the genome aggregation database (gnomAD). 'Other' is the number of heterozygoes diagnosed
        with a cardiac disease which is not LQT3 or BrS1. Click the variant link for more details. Functional
        classification of variants was done according to the following criteria for sodium channel peak tail current,
        stead state voltage of activation, and/or late/persistent current: Loss-of-function (LOF) < 10% peak,
        partial LOF <50% peak, mild LOF 50-75% peak, normal 75-125% peak, Gain-of-function (GOF) > 125% peak current
        or >300% late current, partial LOF >10 mV rightward shift in V<i>0.5</i> activation. Structurally, a "Non_Hotspot"
        variant was surrounded by variants with <10% BrS1 penetrance on average, a "Mild_Hotspot" variant was
        surrounded by variants with between 10% and 40% BrS1 penetrance, on average, a "Hotspot" variant was
        surrounded by variants with greater than 40% BrS1 penetrance, on average.
    </p>
</div>
<!-- Quick Stats Banner - updates with filters -->
<div id="quick-stats" class="my-3 p-3 bg-light rounded-3 text-center" style="display:none;">
    <div class="d-flex flex-wrap gap-3 justify-content-center align-items-center">
        <span class="text-muted">Loading statistics...</span>
    </div>
</div>
    <div class="lds-dual-ring" align="center"></div>
<table id="example" class="table table-striped table-bordered" style="display:none">
    <thead>
    <tr>
        <th>Ch.3</th>
        <th>HGVSc</th>
        <th>Variant</th>
        <th>Residue Number</th>
        <th data-bs-toggle="tooltip" data-bs-placement="top" title="Number of heterozygous carriers diagnosed with Long QT Syndrome type 3">LQT3</th>
        <th data-bs-toggle="tooltip" data-bs-placement="top" title="Number of heterozygous carriers diagnosed with Brugada Syndrome type 1">BrS1</th>
        <th data-bs-toggle="tooltip" data-bs-placement="top" title="Heterozygous carriers from literature and clinical cohorts (excluding gnomAD)">Lit/Cohort</th>
        <th data-bs-toggle="tooltip" data-bs-placement="top" title="Heterozygous carriers from gnomAD population database (assumed unaffected given phenotype rarity)">gnomAD</th>
        <th data-bs-toggle="tooltip" data-bs-placement="top" title="Functional classification based on published electrophysiology studies">Function</th>
        <th data-bs-toggle="tooltip" data-bs-placement="top" title="Structural region within the SCN5A protein">Location</th>
        <th data-bs-toggle="tooltip" data-bs-placement="top" title="Bayesian posterior mean probability of LQT3 diagnosis given heterozygosity">LQT3 Penetrance(%)</th>
        <th data-bs-toggle="tooltip" data-bs-placement="top" title="Bayesian posterior mean probability of Brugada Syndrome type 1 diagnosis given heterozygosity">BrS1 Penetrance(%)</th>
    </tr>
    </thead>
    <tbody>
    <!-- Data loaded via AJAX from /SCN5A/api/variants/ -->
    </tbody>
</table>
<body>
    <div align="center" class="fa-beer"><h1>Histogram of <i>SCN5A</i> Variant Carriers with LQTS</h1></div>
    Below is histogram of the number of heterozygous carriers of <i>SCN5A</i> variants observed in individuals diagnosed
    with LQT3 displayed in the table above. The y-axis is the number of unique variants and the x-axis shows the number of heterozygous carriers.
        <div id='myDiv2'><!-- Plotly chart will be drawn inside this DIV --></div>
    <div class="lds-dual-ring" align="center"></div>
    <div align="center" class="fa-beer"><h1>Histogram of Putative Unaffected Heterozygous <i>SCN5A</i> Variant Carriers</h1></div>
    Below is a histogram of the number of heterozygous carriers of <i>SCN5A</i> variants observed in unaffected
    individuals and in gnomAD (given the rarity of the LQT3 phenotype in the general public, we assume these
    individuals are also likely unaffected) displayed in the table above. The y-axis is the number of unique variants (log scale) and the x-axis
    shows the number of heterozygous carriers (truncated at 300 to aid visualization).
    <div id='myDiv3'><!-- Plotly chart will be drawn inside this DIV --></div>
    <div class="lds-dual-ring" align="center"></div>
    <div align="center" class="fa-beer"><h1>Histogram of <i>SCN5A</i> Variant Carriers with BrS1</h1></div>
    Below is histogram of the number of heterozygous carriers of <i>SCN5A</i> variants observed in individuals diagnosed
    with BrS1 displayed in the table above. The y-axis is the number of unique variants and the x-axis shows the number of heterozygous carriers.
    <div id='myDiv1'><!-- Plotly chart will be drawn inside this DIV --></div>
    <div class="lds-dual-ring" align="center"></div>
    <div align="center" class="fa-beer"><h1>Histogram of BrS1 Penetrance Estimates for <i>SCN5A</i> Variants</h1></div>
    Below is a histogram of the estimates of BrS1 penetrance for all variants displayed in the table above. The range is
    from 0% (impossible) to 100% (certain).
    <div id='myDiv4'><!-- Plotly chart will be drawn inside this DIV --></div>
    <div class="lds-dual-ring" align="center"></div>
    <div align="center" class="fa-beer"><h1>Histogram of LQT3 Penetrance Estimates for <i>SCN5A</i> Variants</h1></div>
    Below is a histogram of the estimates of LQT3 penetrance for all variants displayed in the table above. The range is
    from 0% (impossible) to 100% (certain).
    <div id='myDiv5'><!-- Plotly chart will be drawn inside this DIV --></div>
</body>
{% endblock %}
