{% extends "base.html" %}
{% load static %}
{% block page_content %}

<style>
.lds-dual-ring {
  display: flex;
    justify-content: center;
    align-items: center;
}
.lds-dual-ring:after {
  content: " ";
  display: block;
  width: 64px;
  height: 64px;
  margin: 28px;
  border-radius: 50%;
  border: 6px solid #78cbf1;
  border-color: #646464 transparent #646464 transparent;
  animation: lds-dual-ring 1.2s linear infinite;
}
@keyframes lds-dual-ring {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
div.fa-beer {
    margin-top: 70px;
}
</style>

<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.1.36/pdfmake.min.js"></script>
<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.1.36/vfs_fonts.js"></script>

<script type="text/javascript" src="https://code.jquery.com/jquery-3.5.1.js"></script>
<script type="text/javascript" src="https://cdn.datatables.net/v/ju/dt-1.10.24/b-1.7.0/b-colvis-1.7.0/b-html5-1.7.0/date-1.0.3/r-2.2.7/sc-2.0.3/sb-1.0.1/sl-1.3.3/datatables.min.js"></script>

<script type="text/javascript" src="https://nightly.datatables.net/searchbuilder/js/dataTables.searchBuilder.js"></script>
<script type="text/javascript" src="https://cdn.datatables.net/searchpanes/1.2.1/js/dataTables.searchPanes.min.js"></script>
<script type="text/javascript" src="https://cdn.datatables.net/select/1.3.3/js/dataTables.select.min.js"></script>
<script type="text/javascript" src="https://cdn.datatables.net/buttons/1.7.0/js/dataTables.buttons.min.js"></script>

<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.10.24/css/jquery.dataTables.min.css"/>
<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/searchpanes/1.2.1/css/searchPanes.dataTables.min.css"/>
<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/searchbuilder/1.0.1/css/searchBuilder.dataTables.css"/>
<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/select/1.3.3/css/select.dataTables.min.css"/>
<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/buttons/1.7.0/css/buttons.dataTables.min.css"/>

<script src='https://cdn.plot.ly/plotly-latest.min.js'></script>
<script type="text/javascript" src="{% static 'js/var_table_kcnh2.js' %}"></script>

<div align="center" class="text-center" style="margin-top: 30px">
    <h1>
        <p> <i>KCNH2</i> Variant Browser </p>
    </h1>
</div>
<div>
    <p>
        Below is a comprehensive list of variants found in <i>KCNH2</i> affected or unaffected with the disease most
        commonly associated with variants in <i>KCNH2</i>: long QT type 2 (LQT2; OMIM: 613688). Coding variants are
        referenced to isoform "A" (Q12809-1, transcript ENST00000262186.10) or "B" (Q12809-2, transcript
        ENST00000330883.9) to match convention. Our analysis is primarily for missense and in-frame insertion/deletion
        variants; however, some nonsense variants are included. We do not include synonymous variants.
        'LQT2' is the number of heterozygotes diagnosed with LQT2; 'Unaffected' is the number of heterozygotes
        unaffected with disease, including those found in the genome aggregation database (gnomAD). Click variant link
        for more details. Below are histogram plots for LQT2 affected carriers, unaffected carriers, and estimated LQT2
        penetrance. Use the 'Search Builder' and 'Search Panes' to select subsets of variants based on properties you
        are interested in (results will update the histogram plots below).
    </p>
</div>
    <div class="lds-dual-ring" align="center" ></div>
<table id="example" class="table table-striped table-bordered" style="display:none">
    <thead>
    <tr>
        <th>Ch.7</th>
        <th>Ref</th>
        <th>Alt</th>
        <th>Variant</th>
        <th>HGVSc</th>
        <th>Residue Number</th>
        <th>Native AA</th>
        <th>Variant AA</th>
        <th>LQT2</th>
        <th>Unaffected + gnomAD</th>
        <th>Function</th>
        <th>Location</th>
        <th>LQT2 Penetrance(%)</th>
    </tr>
    </thead>
    <tbody>
    {% for sts in recs %}
    <tr>
        <td>{{sts.pos}}</td>
        <td>{{sts.ref}}</td>
        <td>{{sts.alt}}</td>
        <td><a href="{% url 'kcnh2:detail' sts.hgvsc %}">{{sts.var|truncatechars_html:7}}</a></td>
        <td>{{sts.hgvsc|truncatechars_html:9}}</td>
        <td>{{sts.resnum}}</td>
        <td>{{sts.wtAA}}</td>
        <td>{{sts.mutAA}}</td>
        <td>{{sts.lqt2}}</td>
        <td>{{sts.unaff}}</td>
        <td>{{sts.function}}</td>
        <td>{{sts.structure}}</td>
        <td>{% widthratio sts.p_mean_w 1 100 %}</td>
    </tr>
    {% endfor %}

   </tbody>
</table>
<body>
    <div align="center" class="fa-beer"><h1>Histogram of <i>KCNH2</i> Variant Carriers with LQTS</h1></div>
    Below is histogram of the number of heterozygous carriers of <i>KCNH2</i> variants observed in individuals diagnosed
    with LQT2. The y-axis is the number of unique variants and the x-axis shows the number of heterozygous carriers.
	<div id='myDiv'><!-- Plotly chart will be drawn inside this DIV --></div>
    <div class="lds-dual-ring" align="center" ></div>
    <div align="center" class="fa-beer"><h1>Histogram of Putative Unaffected Heterozygous <i>KCNH2</i> Variant Carriers</h1></div>
    Below is a histogram of the number of heterozygous carriers of <i>KCNH2</i> variants observed in unaffected
    individuals or those in gnomAD (given the rarity of the LQT2 phenotype in the general public, we assume these
    individuals are also likely unaffected). The y-axis is the number of unique variants (log scale) and the x-axis
    shows the number of heterozygous carriers (truncated at 300 to aid visualization).
    <div id='myDiv2'><!-- Plotly chart will be drawn inside this DIV --></div>
    <div class="lds-dual-ring" align="center" ></div>
    <div align="center" class="fa-beer"><h1>Histogram of LQT2 Penetrance Estimates for <i>KCNH2</i> Variants</h1>
    Below is a histogram of the estimates of LQT2 penetrance for all variants displayed in the table above. The range is
    from 0% (impossible) to 100% (certain).
    <div id='myDiv3'><!-- Plotly chart will be drawn inside this DIV --></div>
    <div class="lds-dual-ring" align="center" ></div>
</body>
{% endblock %}