{% extends "base.html" %}
{% load static %}

{% block extra_css %}
    {{ block.super }}
    <style>
    .lds-dual-ring {
      display: flex;
        justify-content: center;
        align-items: center;
    }
    .lds-dual-ring:after {
      content: " ";
      display: block;
      width: 64px;
      height: 64px;
      margin: 28px;
      border-radius: 50%;
      border: 6px solid #78cbf1;
      border-color: #646464 transparent #646464 transparent;
      animation: lds-dual-ring 1.2s linear infinite;
    }
    @keyframes lds-dual-ring {
      0% {
        transform: rotate(0deg);
      }
      100% {
        transform: rotate(360deg);
      }
    }
    div.fa-beer {
        margin-top: 70px;
    }
    </style>
    <link rel="stylesheet" type="text/css" href="{% static 'vendor/datatables/css/jquery.dataTables.min.css' %}"/>
    <link rel="stylesheet" type="text/css" href="{% static 'vendor/datatables/css/searchPanes.dataTables.css' %}"/>
    <link rel="stylesheet" type="text/css" href="{% static 'vendor/datatables/css/searchBuilder.dataTables.css' %}"/>
    <link rel="stylesheet" type="text/css" href="{% static 'vendor/datatables/css/select.dataTables.css' %}"/>
    <link rel="stylesheet" type="text/css" href="{% static 'vendor/datatables/css/buttons.dataTables.css' %}"/>
{% endblock %}

{% block extra_js %}
    {{ block.super }}
    <!-- jQuery already loaded in base.html, don't load again -->
    
    <!-- Load DataTables core -->
    <script type="text/javascript" src="{% static 'vendor/datatables/js/jquery.dataTables.min.js' %}"></script>

    <!-- Load DataTables extensions -->
    <script type="text/javascript" src="{% static 'vendor/datatables/js/dataTables.buttons.min.js' %}"></script>
    <script type="text/javascript" src="{% static 'vendor/datatables/js/dataTables.searchBuilder.min.js' %}"></script>
    <script type="text/javascript" src="{% static 'vendor/datatables/js/dataTables.searchPanes.min.js' %}"></script>
    <script type="text/javascript" src="{% static 'vendor/datatables/js/dataTables.select.min.js' %}"></script>
    <script type="text/javascript" src="{% static 'vendor/datatables/js/dataTables.responsive.min.js' %}"></script>
    
    <!-- Load Plotly -->
    <script src='https://cdn.plot.ly/plotly-2.27.1.min.js'></script>
    
    <!-- Load your custom script last -->
    <script type="text/javascript" src="{% static 'js/var_table_kcnh2.js' %}"></script>
{% endblock %}

{% block page_content %}

<div align="center" class="text-center" style="margin-top: 30px">
    <h1>
        <p> <i>KCNH2</i> Variant Browser </p>
    </h1>
</div>
<div>
    <p>
        Below is a comprehensive list of variants found in <i>KCNH2</i> affected or unaffected with the disease most
        commonly associated with variants in <i>KCNH2</i>: long QT type 2 (LQT2; OMIM: 613688). Coding variants are
        referenced to isoform "A" (Q12809-1, transcript ENST00000262186.10) to match convention.
        Our analysis is primarily for missense and in-frame insertion/deletion
        variants; however, some nonsense variants are included. We do not include synonymous variants.
        'LQT2' is the number of heterozygotes diagnosed with LQT2; 'Unaffected' is the number of heterozygotes
        unaffected with disease, including those found in the genome aggregation database (gnomAD). Multiplexed assay of variant effect (MAVE) data
            demonstrate concentration of the potassium channel at the cell surface, done in very high throughput, referenced to
WT (100%), PMID: 39315434. Click variant link
        for more details. Below are histogram plots for LQT2 affected carriers, unaffected carriers, and estimated LQT2
        penetrance. Use the 'Search Builder' and 'Search Panes' to select subsets of variants based on properties you
        are interested in (results will update the histogram plots below).
    </p>
</div>
    <div class="lds-dual-ring" align="center" ></div>
<table id="example" class="table table-striped table-bordered" style="display:none">
    <thead>
    <tr>
        <th>Ch.7</th>
        <th>HGVSc</th>
        <th>Variant</th>
        <th>Residue Number</th>
        <th>LQT2</th>
        <th>Total Carriers</th>
        <th>MAVE Function</th>
        <th>Location</th>
        <th>LQT2 Penetrance(%)</th>
    </tr>
    </thead>
    <tbody>
    {% for sts in recs %}
    <tr>
        <td>{{sts.pos}}</td>
        <td>{{sts.hgvsc|truncatechars_html:9}}</td>
        <td><a href="{% url 'kcnh2:detail' sts.hgvsc %}" target="_blank">{{sts.var|truncatechars_html:7}}</a></td>
        <td>{{sts.resnum}}</td>
        <td>{{sts.lqt2}}</td>
        <td>{{sts.total_carriers}}</td>
        <td>{{sts.mave_score}}</td>
        <td>{{sts.structure}}</td>
        <td>{% widthratio sts.p_mean_w 1 100 %}</td>
    </tr>
    {% endfor %}

   </tbody>
</table>
<body>
    <div align="center" class="fa-beer"><h1>Histogram of <i>KCNH2</i> Variant Carriers with LQTS</h1></div>
    Below is histogram of the number of heterozygous carriers of <i>KCNH2</i> variants observed in individuals diagnosed
    with LQT2 displayed in the table above. The y-axis is the number of unique variants and the x-axis shows the number of heterozygous carriers.
        <div id='myDiv'><!-- Plotly chart will be drawn inside this DIV --></div>
    <div class="lds-dual-ring" align="center" ></div>
    <div align="center" class="fa-beer"><h1>Histogram of Putative Unaffected Heterozygous <i>KCNH2</i> Variant Carriers</h1></div>
    Below is a histogram of the number of heterozygous carriers of <i>KCNH2</i> variants observed in unaffected
    individuals and in gnomAD (given the rarity of the LQT2 phenotype in the general public, we assume these
    individuals are also likely unaffected) displayed in the table above. The y-axis is the number of unique variants (log scale) and the x-axis
    shows the number of heterozygous carriers (truncated at 300 to aid visualization).
    <div id='myDiv2'><!-- Plotly chart will be drawn inside this DIV --></div>
    <div class="lds-dual-ring" align="center" ></div>
    <div align="center" class="fa-beer"><h1>Histogram of LQT2 Penetrance Estimates for <i>KCNH2</i> Variants</h1></div>
    Below is a histogram of the estimates of LQT2 penetrance for all variants displayed in the table above. The range is
    0-100% penetrance, and we truncated the x-axis at 50% for data presentation. The y-axis is the number of unique
    variants per penetrance bin (0.02 width). Using the 'Search Builder' and 'Search Panes' will filter the histogram
    plots, allowing you to visualize the same subset as in the table above. Selectable panes include a variant's
    observed penetrance (LQT2), structure, functional classification (MAVE surface trafficking), presence of coding
    data, location within the protein and the presence of functional data.
    <div id='myDiv3'><!-- Plotly chart will be drawn inside this DIV --></div>
</body>
{% endblock %}
