
{% extends "base.html" %}
{% load static %}

{% block extra_css %}
    {{ block.super }}
    <style>
    .lds-dual-ring {
      display: flex;
        justify-content: center;
        align-items: center;
    }
    .lds-dual-ring:after {
      content: " ";
      display: block;
      width: 64px;
      height: 64px;
      margin: 28px;
      border-radius: 50%;
      border: 6px solid #78cbf1;
      border-color: #646464 transparent #646464 transparent;
      animation: lds-dual-ring 1.2s linear infinite;
    }
    @keyframes lds-dual-ring {
      0% {
        transform: rotate(0deg);
      }
      100% {
        transform: rotate(360deg);
      }
    }
    div.fa-beer {
        margin-top: 70px;
    }
    </style>
    <!-- DataTables 1.x Core CSS -->
    <link rel="stylesheet" href="https://cdn.datatables.net/1.13.8/css/jquery.dataTables.min.css">
    <!-- Buttons Extension CSS (v2.x for DT1) -->
    <link rel="stylesheet" href="https://cdn.datatables.net/buttons/2.4.2/css/buttons.dataTables.min.css">
    <!-- Select Extension CSS (v1.x for DT1) -->
    <link rel="stylesheet" href="https://cdn.datatables.net/select/1.7.0/css/select.dataTables.min.css">
    <!-- SearchBuilder Extension CSS (v1.x for DT1) -->
    <link rel="stylesheet" href="https://cdn.datatables.net/searchbuilder/1.6.0/css/searchBuilder.dataTables.min.css">
    <!-- SearchPanes Extension CSS (v2.x for DT1) -->
    <link rel="stylesheet" href="https://cdn.datatables.net/searchpanes/2.2.0/css/searchPanes.dataTables.min.css">
{% endblock %}

{% block extra_js %}
    {{ block.super }}
    <!-- DataTables 1.x Core JS -->
    <script src="https://cdn.datatables.net/1.13.8/js/jquery.dataTables.min.js"></script>
    <!-- JSZip (required for Excel export) -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jszip/3.10.1/jszip.min.js"></script>
    <!-- pdfmake (required for PDF export) -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.2.7/pdfmake.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.2.7/vfs_fonts.js"></script>
    <!-- Buttons Extension JS (v2.x for DT1) -->
    <script src="https://cdn.datatables.net/buttons/2.4.2/js/dataTables.buttons.min.js"></script>
    <script src="https://cdn.datatables.net/buttons/2.4.2/js/buttons.html5.min.js"></script>
    <script src="https://cdn.datatables.net/buttons/2.4.2/js/buttons.colVis.min.js"></script>
    <!-- Select Extension JS (v1.x for DT1) -->
    <script src="https://cdn.datatables.net/select/1.7.0/js/dataTables.select.min.js"></script>
    <!-- SearchBuilder Extension JS (v1.x for DT1) -->
    <script src="https://cdn.datatables.net/searchbuilder/1.6.0/js/dataTables.searchBuilder.min.js"></script>
    <!-- SearchPanes Extension JS (v2.x for DT1) -->
    <script src="https://cdn.datatables.net/searchpanes/2.2.0/js/dataTables.searchPanes.min.js"></script>
    <!-- Responsive Extension JS -->
    <script src="https://cdn.datatables.net/responsive/2.5.0/js/dataTables.responsive.min.js"></script>
    <!-- Plotly -->
    <script src='https://cdn.plot.ly/plotly-2.27.1.min.js'></script>
    <!-- Custom KCNQ1 JS -->
    <script type="text/javascript" src="{% static 'js/var_table_kcnq1.js' %}"></script>
{% endblock %}

{% block page_content %}

<div align="center" class="text-center" style="margin-top: 30px">
    <h1>
        <p> <i>KCNQ1</i> Variant Browser </p>
    </h1>
</div>
<div>
    <p>
        Here we have a comprehensive list of variants in <i>KCNQ1</i> found in individuals affected and unaffected with the
        disease most commonly associated with variants in KCNQ1, congenital long QT type 1 (LQT1; OMIM: 192500). Coding
        variants are referenced to isoform P51787 (ENST00000155840.12) to
        match convention. 'Total Carriers' is the number of heterozygotes diagnosed with LQT1 or other disease,
        Unaffected with disease, and found in the genome aggregation database (gnomAD). 'Other' is the number of
        heterozygoes diagnosed with a cardiac disease which is not LQT1. Click the variant link for more details.
        Below are histogram plots for LQT2 affected carriers, unaffected carriers, and estimated LQTS
        penetrance. Use the 'Search Builder' and 'Search Panes' to select subsets of variants based on properties you
        are interested in (results will update the histogram plots below). 'GOF' and 'LOF' indicate gain-of-function and
        loss-of-function, respectively.
    </p>
</div>
    <div class="lds-dual-ring" align="center"></div>
<table id="example" class="table table-striped table-bordered" style="display:none">
    <thead>
    <tr>
        <th>Ch.11</th>
        <th>HGVSc</th>
        <th>Variant</th>
        <th>Residue Number</th>
        <th>LQT1</th>
        <th>Total Carriers</th>
        <th>Function</th>
        <th>Location (LQT1)</th>
        <th>LQT1 Penetrance(%)</th>
    </tr>
    </thead>
    <tbody>
    {% for sts in recs %}
    <tr>
        <td>{{sts.pos}}</td>
        <td>{{sts.hgvsc|truncatechars_html:12}}</td>
        <td><a href="{% url 'kcnq1:detail' sts.hgvsc %}" target="_blank">{{sts.var|truncatechars_html:7}}</a></td>
        <td>{{sts.resnum}}</td>
        <th>{{sts.lqt1}}</th>
        <td>{{sts.total_carriers}}</td>
        <td>{{ sts.function_lqt1 }}</td>
        <td>{{sts.structure_lqt1}}</td>
        <td>{% widthratio sts.lqt1_penetrance 1 100 %}</td>
    </tr>
    {% endfor %}

   </tbody>
</table>
    <body>
    <div align="center" class="fa-beer"><h1>Histogram of <i>KCNQ1</i> Variant Carriers with LQTS</h1></div>
    Below is histogram of the number of heterozygous carriers of <i>KCNQ1</i> variants observed in individuals diagnosed
    with LQT1 displayed in the table above. The y-axis is the number of unique variants and the x-axis shows the number of heterozygous carriers.
	<div id='myDiv'><!-- Plotly chart will be drawn inside this DIV --></div>
    <div class="lds-dual-ring" align="center" ></div>
    <div align="center" class="fa-beer"><h1>Histogram of Putative Unaffected Heterozygous <i>KCNQ1</i> Variant Carriers</h1></div>
    Below is a histogram of the number of heterozygous carriers of <i>KCNQ1</i> variants observed in unaffected
    individuals and in gnomAD (given the rarity of the LQT1 phenotype in the general public, we assume these
    individuals are also likely unaffected) displayed in the table above. The y-axis is the number of unique variants (log scale) and the x-axis
    shows the number of heterozygous carriers (truncated at 300 to aid visualization).
    <div id='myDiv2'><!-- Plotly chart will be drawn inside this DIV --></div>
    <div class="lds-dual-ring" align="center" ></div>
    <div align="center" class="fa-beer"><h1>Histogram of LQT1 Penetrance Estimates for <i>KCNQ1</i> Variants</h1></div>
    Below is a histogram of the estimates of LQT1 penetrance for all variants displayed in the table above. The range is
    from 0% (impossible) to 100% (certain).
    <div id='myDiv3'><!-- Plotly chart will be drawn inside this DIV --></div>
    <div class="lds-dual-ring" align="center" ></div>
</body>

{% endblock %}